pandas/core/arrays/arrow/dtype.py

```diff
@@ -90,7 +100,36 @@ def type(self):
         """
         Returns pyarrow.DataType.
         """
-        return type(self.pyarrow_dtype)
+        pa_type = self.pyarrow_dtype
+        if pa.types.is_integer(pa_type):
+            return int
+        elif pa.types.is_floating(pa_type):
+            return float
+        elif pa.types.is_string(pa_type):
+            return str
+        elif pa.types.is_binary(pa_type):
+            return bytes
+        elif pa.types.is_boolean(pa_type):
+            return bool
+        elif pa.types.is_duration(pa_type):
+            if pa_type.unit == "ns":
+                return Timedelta
+            else:
+                return timedelta
+        elif pa.types.is_timestamp(pa_type):
+            if pa_type.unit == "ns":
+                return Timestamp
+            else:
+                return datetime
+        elif pa.types.is_date(pa_type):
+            return date
+        elif pa.types.is_time(pa_type):
+            return time
+        elif pa.types.is_null(pa_type):
+            # TODO: None? pd.NA? pa.null?
```

Comment:
It _seems_ like this should be `None` since we want to prefer the python scalar?

Reply:
i guess that depends on what type is supposed to mean here.  e.g. should we expect `isinstance(obj[0], obj.dtype.type)`?  if so, we'd need something like `type(None)` rather than `None` itself

Reply:
In that case it's actually `type(pd.NA)`, since getting a scalar gives NA and not None